pastis/temporal_analysis/harris_mode.ipynb

```diff
@@ -0,0 +1,3338 @@
+   "source": [
+    "start_time = time.time()\n",
+    "focus_fieldS = []\n",
+    "focus_fieldS_Re = []\n",
+    "focus_fieldS_Im = []\n",
+    "\n",
+    "for pp in range(0, n_harris):\n",
+    "    print(pp)\n",
+    "    harris_modes = np.zeros(n_harris)\n",
+    "    harris_modes[pp] = (nm_aber) / 2\n",
+    "    luvoir.harris_sm.actuators  = harris_modes\n",
+    "    harris_meas = luvoir.calc_out_of_band_wfs(norm_one_photon=True)\n",
+    "    harris_meas_sub_real = hcipy.field.subsample_field(harris_meas.real, z_pup_downsample, grid_zernike, statistic='mean')\n",
+    "    harris_meas_sub_imag = hcipy.field.subsample_field(harris_meas.imag, z_pup_downsample, grid_zernike, statistic='mean')\n",
+    "    focus_field1 = harris_meas_sub_real + 1j * harris_meas_sub_imag\n",
+    "    focus_fieldS.append(focus_field1)\n",
+    "    focus_fieldS_Re.append(focus_field1.real)\n",
+    "    focus_fieldS_Im.append(focus_field1.imag)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "alternative-chinese",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "filename_matrix = 'EFIELD_OBWFS_Re_matrix_num_harris_' + str(n_harris)\n",
+    "hcipy.write_fits(focus_fieldS_Re, os.path.join(resDir, filename_matrix + '.fits'))\n",
+    "print('Efield Real saved to:', os.path.join(resDir, filename_matrix + '.fits'))\n",
+    "\n",
+    "filename_matrix = 'EFIELD_OBWFS_Im_matrix_num_harris_' + str(n_harris)\n",
+    "hcipy.write_fits(focus_fieldS_Im, os.path.join(resDir, filename_matrix + '.fits'))\n",
+    "print('Efield Imag saved to:', os.path.join(resDir, filename_matrix + '.fits'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "planned-browse",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "filename_matrix = 'EFIELD_OBWFS_Re_matrix_num_harris_' + str(n_harris)+'.fits'\n",
+    "G_OBWFS_real = fits.getdata(os.path.join(overall_dir, 'matrix_numerical', filename_matrix))\n",
+    "filename_matrix = 'EFIELD_OBWFS_Im_matrix_num_harris_' + str(n_harris)+'.fits'\n",
+    "G_OBWFS_imag =  fits.getdata(os.path.join(overall_dir, 'matrix_numerical', filename_matrix))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "mineral-murray",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "G_OBWFS= np.zeros([N_pup_z*N_pup_z,2,n_harris])\n",
+    "for pp in range(0, n_harris):\n",
+    "    G_OBWFS[:,0,pp] = G_OBWFS_real[pp]*z_pup_downsample - Efield_ref_OBWFS.real\n",
+    "    G_OBWFS[:,1,pp] = G_OBWFS_imag[pp]*z_pup_downsample - Efield_ref_OBWFS.imag"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "governmental-skill",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "def req_closedloop_calc_recursive(Gcoro, Gsensor, E0coro, E0sensor, Dcoro, Dsensor, t_exp, flux, Q, Niter, dh_mask,\n",
```

Comment:
This should be imported from 
https://github.com/spacetelescope/PASTIS/blob/dcd51b51fa3fa64c88e492b857623e49a97fc7e7/pastis/temporal_analysis/close_loop_analysis.py#L3
that's why we put it there. When you change this, make sure it still runs correctly please.